Lib/test/test_cmd_line.py

```diff
@@ -579,13 +579,13 @@ def test_unknown_options(self):
                      'Cannot run -I tests when PYTHON env vars are required.')
     def test_isolatedmode(self):
         self.verify_valid_flag('-I')
-        self.verify_valid_flag('-IEs')
+        self.verify_valid_flag('-IEPs')
         rc, out, err = assert_python_ok('-I', '-c',
             'from sys import flags as f; '
-            'print(f.no_user_site, f.ignore_environment, f.isolated)',
+            'print(f.no_user_site, f.ignore_environment, f.isolated, f.safe_path)',
             # dummyvar to prevent extraneous -E
             dummyvar="")
-        self.assertEqual(out.strip(), b'1 1 1')
+        self.assertEqual(out.strip(), b'1 1 1 True')
```

Comment:
Curious question. Why is `safe_path` True and not 1?

Reply:
In the past, sys.flags values were numbers because it was trivial to increment global configuration variables like Py_IgnoreEnvironmentFlag when parsing the command line. sys.flags just exposed these raw values. The API was driven by "it is easy to implement". But to me, it makes no sense that sys.flags.ignore_environment can be equal to 2 to 1000.

Example with Python 2:

```
$ python2 -EEEEEEEEEEE -c 'import sys; print(sys.flags.ignore_environment)'
11
```

It makes sense for the -O option where 1 and 2 have different meanings:

```
$ ./python -OO -c 'import sys; print(sys.flags.optimize)'
2
```

But... Python accepts higher values even if it makes no sense (maybe it should be fixed, but I didn't change that in PEP 587):

```
$ ./python -OOOOOOOOOOOOOOOOO -c 'import sys; print(sys.flags.optimize)'
17
```


When I implemented PEP 587, I made it more reasonable:

```
$ python3.10 -EEEEEEEEEEE -c 'import sys; print(sys.flags.ignore_environment)'
1
```

For new options like sys.flags.utf8_mode, I prefer a boolean which IMO makes more sense.